components/enterprise/subscriptions.tf

```diff
@@ -1,62 +1,37 @@
-
-  platform                = { for k, v in var.platform_subscriptions : k => merge(tomap({ group = "platform" }), v) }
-  platform_sandbox        = { for k, v in var.platform_sandbox_subscriptions : k => merge(tomap({ group = "platform_sandbox" }), v) }
-  platform_non_production = { for k, v in var.platform_non_production_subscriptions : k => merge(tomap({ group = "platform_non_production" }), v) }
-  platform_production     = { for k, v in var.platform_production_subscriptions : k => merge(tomap({ group = "platform_production" }), v) }
-
-  subscriptions = merge(
-    local.cft,
-    local.cft_sandbox,
-    local.cft_non_production,
-    local.cft_production,
-
-    local.sds,
-    local.sds_sandbox,
-    local.sds_non_production,
-    local.sds_production,
-
-    local.crime,
-
-    local.heritage,
-    local.heritage_sandbox,
-    local.heritage_non_production,
-    local.heritage_production,
-
-    local.security,
-
-    local.platform,
-    local.platform_sandbox,
-    local.platform_non_production,
-    local.platform_production,
-  )
-}
-
 module "subscription" {
   for_each = local.subscriptions
 
-  source = "../../modules/subscription"
-  name   = each.key
-  value  = each.value
+  source      = "../../modules/subscription"
+  name        = each.key
+  value       = each.value
+  common_tags = module.tags[each.key].common_tags
+  environment = try(each.value.environment, (lower(replace([each.key][0], "/.*-([A-Za-z]+).*/", "$1"))))
 
   billing_account_name    = var.billing_account_name
   enrollment_account_name = var.enrollment_account_name
 }
+
+module "acme" {
+  for_each = { for k, v in local.subscriptions : k => v if try(v.deploy_acme, false) }
+
+  source = "git::https://github.com/hmcts/terraform-module-acme-function.git?ref=DTSPO-9743/ops-bootstrap"
+
+  location                       = var.location
+  env                            = try(each.value.environment, (lower(replace([each.key][0], "/.*-([A-Za-z]+).*/", "$1"))))
+  common_tags                    = module.tags[each.key].common_tags
+  dns_contributor_group_id       = module.environment[try(each.value.environment, (lower(replace([each.key][0], "/.*-([A-Za-z]+).*/", "$1"))))].dns_contributor_group_id
+  product                        = var.product
+  resource_group_name            = module.environment[try(each.value.environment, (lower(replace([each.key][0], "/.*-([A-Za-z]+).*/", "$1"))))].resource_group_name
+  resource_group_id              = module.environment[try(each.value.environment, (lower(replace([each.key][0], "/.*-([A-Za-z]+).*/", "$1"))))].resource_group_id
+  asp_id                         = module.environment[try(each.value.environment, (lower(replace([each.key][0], "/.*-([A-Za-z]+).*/", "$1"))))].asp_id
+  subscription_id                = module.subscription[each.key].subscription_id.subscription_id
+  acme_storage_account_repl_type = try(each.value.acme_storage_account_repl_type, "ZRS")
```

Comment:
do we still need this? I assume it's legacy based on transition?

Reply:
Yeah, there's a couple still on LRS